src/hackerspace_online/middleware.py

```diff
@@ -10,3 +13,29 @@ def __call__(self, request):
         connections['default'].force_debug_cursor = True
         response = self.get_response(request)
         return response
+
+
+class RequestDataTooBigMiddleware:
+    """
+    Custom middleware to handle requests that exceed the settings.DATA_UPLOAD_MAX_MEMORY_SIZE
+    """
+    def __init__(self, get_response):
+        self.get_response = get_response
+
+    def __call__(self, request):
+        """
+        If the request body is too big then request.body will throw a
+        RequestDataTooBig exception. Check here explicitly if that will happen.
+        """
+        try:
+            _ = request.body
+        except RequestDataTooBig:
+            # if the size of the request (excluding any file uploads) exceeded settings.DATA_UPLOAD_MAX_MEMORY_SIZE.
+
+            # generate a more sensible response and redirect back
+            messages.add_message(
+                request, messages.ERROR, "The size of the request is too large. Please reduce its size and try again.")
```

Comment:
Let's use this error (Do template filters work in messages?)

"This requests exceeds the maximum size of {{settings.DATA_UPLOAD_MAX_MEMORY_SIZE|filesizeformat}}.  This is likely caused by one of the text fields contains too much data, either from text (including whitespace) or a large data URI (an image or other data encoded as text)."

Reply:
done, more verbose error message added in https://github.com/bytedeck/bytedeck/pull/1455/commits/345b484ecf4c1911f2923b7b08466ad2598f2415